storage/mongodb/files/scripts/test-mongo.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+
+readonly NAMESPACE=${1}
+readonly ITERATION=${2}
+readonly THREADS=${3}
+readonly TMP_FILE=${4}
+
+readonly MONGODB_IP=$(oc get svc -n ${NAMESPACE} | grep mongodb | awk '{print $3}')
+
+if [[ ! -z "${benchmark_results_dir}" ]]; then
+ echo "${benchmark_results_dir}" >> "${TMP_FILE}"
+fi
+
+echo "NAMESPACE: ${NAMESPACE}"
+echo "ITERATION: ${ITERATION}"
+echo "THREADS: ${THREADS}"
+echo "TMP_FILE: ${TMP_FILE}"
+
+for i in $(seq 1 ${ITERATION});
+do
+  echo "iteration: ${i}"
+  oc exec $(oc get pod -n ${NAMESPACE} | grep mongodb | awk '{print $1}') -- scl enable rh-mongodb32 -- mongo -u redhat -p redhat ${MONGODB_IP}:27017/testdb --eval "db.usertable.remove({})"
+  oc exec $(oc get pod -n ${NAMESPACE} | grep ycsb | awk '{print $1}') -- ./bin/ycsb load mongodb -s -threads "${THREADS}" -P workloads/workload_template -p mongodb.url=mongodb://redhat:redhat@${MONGODB_IP}:27017/testdb | egrep 'RunTime|Throughput' | cut -d',' -f2,3 > $benchmark_run_dir/mongodb_result_run_$i.txt
```

Comment:
I am proposing to have results saved *also* in $benchmark_run_dir ... comment?

Reply:
@hongkailiu yctb allows to run different loads eg, workloada , workloadb ... etc , maybe we can adapt this script to something like 
https://gist.github.com/ekuric/089b2a6e19350b2c2990a2b43661b411 
then we can external_vars.yaml specify what load we want eg. worklaoda , workloadb, or all of them. new variable in external_vars.yaml would be 
``` workload: workloada,workloadb,workloadc,workloadd,workloade,workloadf,workload_template``` 

in mongodb-test.yaml will be necessary to add new variable workload 
```  command: pbench-user-benchmark --config="mongodb_storage_test_{{ STORAGE_CLASS_NAME }}" -- "{{ tmp_folder }}/files/scripts/test-mongo.sh" "{{ test_project_name }}" "{{ iteration }}" "{{ ycsb_threads }}" "{{ tmp_file_result.path }}" "{{ workload }}" ``` 

comments welcome!

Reply:
Done